### ISSUE TITLE
Date-Dropdown with dates from Google Calendar

### ISSUE THREAD

**Issue by yelch** [tlovett1/custom-contact-forms#228]
Not an issue, but could be used as an enhancement and the reason for my previous request https://github.com/tlovett1/custom-contact-forms/issues/209
--> with such a field all code could be kept in the form

Once a month our club offers free trials, but people should signup via a form. 
We keep those dates in a public google calendar beside other events. So I needed only those eventdates containing the subject 'trial' (in events like 'Training with trials') to be shown as dropdown in a form. I managed this with custom-contact-forms and a bit of jquery:

Create a normal dropdown, name it f.e. 'MyDate' (slug, not label). Then create a Page, insert the Form and in Source-View insert the following code at the bottom:
```javascript
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="http://www.google.com/jsapi?autoload=%7Bmodules%3A%5B%7Bname%3Agdata%2Cversion%3A1.x%2Cpackages%3A%5Bcalendar%5D%7D%5D%7D"></script>
<script type="text/javascript">
<!--
!function(e) {
    e.GoogleCalendarEvents = function(n) {
        function l(l) {
            var a = l.items;
            e($("select[name='ccf_field_" + n.element + "']")).html("");
            //Create google event list
            for (var s = null, g = 0; g < a.length; g++) {
				var i = a[g], r = i.summary, p = new Date(i.start.dateTime), myDate = "";
                               if (isNaN(p.getDay())) {}
			       else if (r.indexOf( n.findstring ) < 1) {} 
                               else {
					var d = p.getDate(), m = p.getMonth()+1, y = p.getFullYear()
					myDate = d + "." + m + "." + y;
					e($("select[name='ccf_field_" + n.element + "']")).append('<option value="' + myDate + '">' + myDate + '</option>');
				}
            }
        }
        var a = {
            displayCount: !0,
            dateLabel: "",
            groupDates: !1
        };
        n = e.extend(a, n);
        var s = new Date().toISOString();
        e.get("https://www.googleapis.com/calendar/v3/calendars/" + n.calendarId + "/events", {
            maxResults: n.maxResults,
            singleEvents: !0,
            orderBy: "startTime",
            timeMin: s,
            key: n.apiKey
        }).done(function(e) {
            l(e);
        });
    };
}(jQuery);
$.GoogleCalendarEvents({ apiKey: "YOURAPIKEY", calendarId: 'YOURCALENDARID', element: 'MyDate', maxResults: 30, findstring: 'YOUR_STRING' });
//-->
</script>
```
Adjust the apiKey (see f.e. here: http://fullcalendar.io/docs/google_calendar/ ), calendarId ('emailaddress' of your calendar), element (=fieldname), maxResults (depends on your calendar), findstring and perhaps your desired date format in the 'myDate' line
If every date of a specific calendar should be listed it's even simpler: just remove the line
```javascript
else if (r.indexOf( n.findstring ) < 1) {} 
```
Hope this helps somebody extending this great tool.
